lib/pbench/server/api/__init__.py

```diff
@@ -17,41 +18,69 @@
 from pbench.common.logger import get_pbench_logger
 from pbench.server.api.resources.query_apis.elasticsearch_api import Elasticsearch
 from pbench.server.api.resources.query_apis.query_controllers import QueryControllers
+from pbench.server.database.database import Database
 from pbench.server.api.resources.query_apis.query_month_indices import QueryMonthIndices
+from pbench.server.api.auth import Auth
+
+from pbench.server.api.resources.pbench_users import (
+    RegisterUser,
+    LoginAPI,
+    UserAPI,
+)
 
 
 def register_endpoints(api, app, config):
     """Register flask endpoints with the corresponding resource classes
     to make the APIs active."""
 
     base_uri = config.rest_uri
-    app.logger.info("Registering service endpoints with base URI {}", base_uri)
+    logger = app.logger
+
+    # Init the the authentication module
+    token_auth = Auth()
+    Auth.set_logger(logger)
+
+    logger.info("Registering service endpoints with base URI {}", base_uri)
 
     api.add_resource(
         Upload,
         f"{base_uri}/upload/ctrl/<string:controller>",
-        resource_class_args=(config, app.logger),
+        resource_class_args=(config, logger),
     )
     api.add_resource(
-        HostInfo, f"{base_uri}/host_info", resource_class_args=(config, app.logger),
+        HostInfo, f"{base_uri}/host_info", resource_class_args=(config, logger),
     )
     api.add_resource(
         Elasticsearch,
         f"{base_uri}/elasticsearch",
-        resource_class_args=(config, app.logger),
+        resource_class_args=(config, logger),
     )
     api.add_resource(
-        GraphQL, f"{base_uri}/graphql", resource_class_args=(config, app.logger),
+        GraphQL, f"{base_uri}/graphql", resource_class_args=(config, logger),
     )
     api.add_resource(
         QueryControllers,
         f"{base_uri}/controllers/list",
-        resource_class_args=(config, app.logger),
+        resource_class_args=(config, logger),
     )
     api.add_resource(
         QueryMonthIndices,
         f"{base_uri}/controllers/months",
-        resource_class_args=(config, app.logger),
+        resource_class_args=(config, logger),
+    )
+
+    api.add_resource(
+        RegisterUser, f"{base_uri}/user", resource_class_args=(config, logger),
+    )
+    api.add_resource(
+        LoginAPI,
+        f"{base_uri}/session",
```

Comment:
Let's find another name besides `session`, as the terminology for "seesion" is browsers has to do with tracking the instance of a browser talking to a server, and typically not a logged in user.

Perhaps consider having separate endpoints for logging in, `/login` and logging out, `/logout`.

Reply:
The terminology varies widely; but again, a "user session" (whatever we call it) is a server resource that is created (POST) and deleted (DELETE), and I'd prefer not to use separate URIs. Unfortunately, the Flask methodology makes this a bit messier than it should be because we declare the basic URI here, and the operations are in the resource class so they're not obvious here. I think that using Flask blueprints might make this a bit neater, and both Nikhil and I have thought about looking into that.

Reply:
Why not `/user/<name>/login` `POST` to log in, and `DELETE` on the same to log out?

Reply:
That's certainly possible, although a bit odd. The username has to be a parameter anyway.